httpblock.go

```diff
@@ -338,7 +339,7 @@ func newHttpBlock(opts *warcRecordOptions, r io.Reader, blockDigest, payloadDige
 		rb = bufio.NewReader(r)
 	}
 
-	b, err := rb.Peek(4)
+	_, err := rb.Peek(4)
 	if err != nil {
 		return nil, fmt.Errorf("not a http block: %w", err)
 	}
```

Comment:
Could this code be deleted? Are there cases where this would catch an error which is not caught by headerBytes()?

Reply:
We initially removed this, but the argument to keep it was that it is a sanity check that the buffer contains at least 4 bytes. But yes, it can be removed.

Reply:
It can be configusing to trace the error if it is from headerBytes or even more configusing if the error is returned from parsing below (as a httpRequestBlock).

Reply:
It might be an idea to move the code to the headerBytes function if it is kept since it peeks at headerBytes anyways

Reply:
I think the check should be moved to headerBytes() function.

Reply:
Moved peek and added a small comment that hopefully is precise